### ISSUE TITLE
Undefined index: argv in {closure}() (line 19 of /usr/share/drupal7/psysh)

### ISSUE THREAD

**Issue by Lucas-C** [bobthecow/psysh#237]
Hi,

First, a giant THANK YOU: I come from the Python dev world, and this tool is awesome.

Now, I have a small bug report for you. Here is the error I get in Drupal:
```
Notice: Undefined index: argv in {closure}() (line 19 of /usr/share/drupal7/psysh).
Warning: Invalid argument supplied for foreach() in {closure}() (line 19 of /usr/share/drupal7/psysh).
```

It is probably self-explanatory (this code should probably not assume `argv` is always present in `$_SERVER`), but don't hesitate to ask if you need more info

**Comment by bobthecow** [comment]
I believe `argv` _should_ always be present in `$_SERVER`. How are you running it?

**Comment by Lucas-C** [comment]
What do you mean ?
It's served by Apache

**Comment by bobthecow** [comment]
I mean how are you invoking PsySH?

**Comment by Lucas-C** [comment]
From memory:
```
require DRUPAL_ROOT . "/psysh";
eval(\Psy\sh());
```

**Comment by bobthecow** [comment]
And that code is being run in a web request via Apache?

**Comment by Lucas-C** [comment]
Yes, the error appeared when I left this snippet in code.

**Comment by daang77** [comment]
@Lucas-C I've had a similar problem with my setup, and it turned out my php.ini was not correctly set up. I had to enable register_argc_argv=on .. and then the error message disappeared. Found the solution here: http://stackoverflow.com/questions/14108592/composer-not-installing-uncaught-exception-errorexception-with-message-undefi

Other than that I believe you have to run your app with the built in php sever using 'php -S' in order to drop to the shell .. more info here http://php.net/manual/en/features.commandline.webserver.php

I personally am not able to run my app this way easily since much of the include paths and aliases happen at Apache level, but at least the 'argv' error is solved for me :)

**Comment by Lucas-C** [comment]
Thanks @daang77 !
Nice of you to share this solution here

**Comment by daang77** [comment]
@Lucas-C No problem, that's what GitHub is for ;) 

I hope it will work for you.

**Comment by bobthecow** [comment]
Sorry about all that. I think I've got a fix in 62f59b66

**Comment by RayFoss** [comment]
I can confirm both solutions work to remove the error. that is to say making `register_argc_argv=on` in php.ini or getting the latest dev version. `composer g require dev-feature/runkit-redux`

For some reason the require outputs #!/usr/bin/env php everywhere. I'll have to output stream that for now.
